x/ethbridge/types/msgs.go

```diff
@@ -12,28 +12,58 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+type MessageType uint32
+
+const (
+	MsgSubmit          MessageType = 0
+	MsgRevert          MessageType = 1
+	MsgReturnCeth      MessageType = 2
+	InvalidMessageType MessageType = 3
+)
+
+// MessageTypeStringToEnum return MessageType according to input string
+func MessageTypeStringToEnum(messageType string) MessageType {
+	switch messageType {
+	case "submit":
+		return MsgSubmit
+	case "revert":
+		return MsgRevert
+	case "return":
+		return MsgReturnCeth
+	}
+	return InvalidMessageType
+}
+
 // MsgLock defines a message for locking coins and triggering a related event
 type MsgLock struct {
-	CosmosSender     sdk.AccAddress  `json:"cosmos_sender" yaml:"cosmos_sender"`
-	Amount           sdk.Int         `json:"amount" yaml:"amount"`
-	Symbol           string          `json:"symbol" yaml:"symbol"`
-	EthereumChainID  int             `json:"ethereum_chain_id" yaml:"ethereum_chain_id"`
-	EthereumReceiver EthereumAddress `json:"ethereum_receiver" yaml:"ethereum_receiver"`
+	CosmosSender         sdk.AccAddress  `json:"cosmos_sender" yaml:"cosmos_sender"`
+	CosmosSenderSequence uint64          `json:"cosmos_sender_sequence" yaml:"cosmos_sender_sequence"`
+	Symbol               string          `json:"symbol" yaml:"symbol"`
+	Amount               sdk.Int         `json:"amount" yaml:"amount"`
+	EthereumChainID      int             `json:"ethereum_chain_id" yaml:"ethereum_chain_id"`
+	CethAmount           sdk.Int         `json:"ceth_amount" yaml:"ceth_amount"`
+	MessageType          MessageType     `json:"message_type" yaml:"message_type"`
+	EthereumReceiver     EthereumAddress `json:"ethereum_receiver" yaml:"ethereum_receiver"`
 }
 
 // NewMsgLock is a constructor function for MsgLock
-func NewMsgLock(
-	ethereumChainID int, cosmosSender sdk.AccAddress,
-	ethereumReceiver EthereumAddress, amount sdk.Int, symbol string) MsgLock {
+func NewMsgLock(ethereumChainID int, cosmosSender sdk.AccAddress, ethereumReceiver EthereumAddress, amount sdk.Int,
+	symbol string, cethAmount sdk.Int, messageType MessageType) MsgLock {
 	return MsgLock{
-		EthereumChainID:  ethereumChainID,
-		CosmosSender:     cosmosSender,
-		EthereumReceiver: ethereumReceiver,
-		Amount:           amount,
-		Symbol:           symbol,
+		EthereumChainID:      ethereumChainID,
+		CosmosSender:         cosmosSender,
+		CosmosSenderSequence: 0,
```

Comment:
We can't have the cosmos sender sequence set to 0. This value must be passed in as a param. I don't see anywhere that it is set later to the correct sequence.